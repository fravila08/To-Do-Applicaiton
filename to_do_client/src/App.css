body {
  width: 100vw;
  height: 100vh;
}

.App {
  width: 100vw;
  height: 100vh;
  margin: 0;
}

h1 {
  color: black;
  font-weight: 300;
  text-shadow: 0em;
  text-decoration: underline;
}

#Header {
  text-align: center;
}

/* List holders [pending/completed] */

.listHolder {
  border: black solid 4px;
  min-height: 40vh;
  background-color: lightgrey;
  text-align: center;
  padding: 0 5vw;
  display: flex;
  flex-direction: column;
  align-items: center;
  overflow: auto;
  height: 10vh;
}

#CompletedList {
  border-radius: 0 0 10vw 10vw;
}

#pending {
  border-radius: 10vw 10vw 0 0;
}

.listHolder::-webkit-scrollbar {
  display: none;
}

.listHeader {
  background-color: whitesmoke;
  border-radius: 5vw;
  margin-top: 1vh;
  padding: 1vh 10vw;
  text-align: center;
  width: 50vw;
}

/* Task Row */

.task {
  background-color: grey;
  border: black 2px solid;
  height: 4vh;
  width: 80vw;
  border-radius: 1vh;
}

.taskTitle {
  display: flex;
  text-align: left;
  font-size: 2vh;
}

.checkHolder {
  display: flex;
  justify-content: center;
  align-items: top;
  position: relative;
}
.check {
  position: absolute;
  opacity: 0;
  z-index: 2;
}

.checkImg {
  height: 3vh;
  position: absolute;
  z-index: 1;
}

/* Input field on upper right hand corner */

.FormHolder {
  margin: 0;
  padding: 0;
}

Form {
  display: flex;
  flex-direction: row-reverse;
}

#createTaskInput {
  font-size: 2vh;
  max-width: 50vw;
}

#createTaskButton {
  border-radius: 0 1vh 1vh 0;
  height: 5vh;
  position: absolute;
  right: 0;
  display: flex;
  justify-content: center;
  align-items: center;
}
