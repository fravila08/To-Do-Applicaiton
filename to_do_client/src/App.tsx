import React, {  useEffect, useState } from 'react'
import './App.css'
import Container from 'react-bootstrap/Container'
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'
import { Task } from './components/Task'
import axios from 'axios'
import { CsrfToken } from './components/CsrfToken'

export interface TaskInterface {
    id: number;
    Title: string;
    Completed:boolean;
}

export interface ResponseTasks{
    completed: TaskInterface[];
    pending: TaskInterface[];
    biggestID: number;
}

export async function getTasks():Promise<ResponseTasks>{
    let response = await axios.get('allTasks')
    return response.data
}


function App() {
  
  const [compTasks, setCompTasks] : [TaskInterface[] , (compTasks:TaskInterface[])=>void]=useState<TaskInterface[]>([])
  const [pendTasks, setPendTasks] : [TaskInterface[] , (pendTasks:TaskInterface[])=>void]=useState<TaskInterface[]>([])
  
  useEffect(()=>{
      const getResponse=async ()=>{ 
          let response=await getTasks()
          setCompTasks(response.completed)
          setPendTasks(response.pending)
      }
      getResponse()
  },[])
  
  CsrfToken()

  return (
    <Container className="App"> 

      <Row id='Header'>
          <h1>To Do App</h1>
      </Row>

      <Row>

        <Col xs={1}></Col>

        <Col xs={10} className='listHolder'>
          
          <h5 className='listHeader'> Pending </h5>
          
          {pendTasks.map((task)=>(
            <Task 
              task={task} 
            />
          ))}

        </Col>

        <Col xs={1}></Col>

      </Row>

      <Row>

        <Col xs={1}></Col>

        <Col xs={10} className='listHolder'>
          
          <h5 className='listHeader'> Completed </h5>
        
          {compTasks.map((task)=>(
            <Task 
              task={task}
            />
          ))}
      
        </Col>

        <Col xs={1}></Col> 

      </Row>
    </Container>
  )
}

export default App
