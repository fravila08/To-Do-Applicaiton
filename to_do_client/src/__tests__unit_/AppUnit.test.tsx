import { describe, expect, it, vi, Mocked } from "vitest";
import TestRenderer from "react-test-renderer";
import axios from "axios";
import { getTasks } from "../App";
import App from "../App";

vi.mock("axios");

describe("App", () => {
  describe("getTasks()", () => {
    it("will return an array of tasks", async () => {
      const mockedAxios = axios as Mocked<typeof axios>;
      mockedAxios.get.mockResolvedValue({ data: { tasks: [] } });
      const tasks = await getTasks();
      expect(tasks).toStrictEqual([]);
    });
  });

  it("will create and match snapshot", () => {
    const myApp = TestRenderer.create(<App />);
    expect(myApp).toMatchSnapshot();
  });
});
