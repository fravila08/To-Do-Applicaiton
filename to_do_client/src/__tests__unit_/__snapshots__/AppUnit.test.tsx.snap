// Vitest Snapshot v1

exports[`App > will create and match snapshot 1`] = `
<div
  className="App container"
>
  <div
    className="row"
    id="Header"
  >
    <h1>
      To Do App
    </h1>
  </div>
  <div
    className="row"
  >
    <div
      className="container"
      style={
        {
          "marginBottom": "1vh",
        }
      }
    >
      <div
        className="row"
      >
        <div
          className="col-4"
        >
          <button
            className="btn btn-primary"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            CS
          </button>
        </div>
        <div
          className="formHolder col-8"
        >
          <form
            className=""
            onSubmit={[Function]}
          >
            <input
              className="form-control"
              id="createTaskInput"
              onChange={[Function]}
              value=""
            />
            <button
              className="btn btn-success"
              disabled={true}
              id="createTaskButton"
              type="submit"
            >
              Create
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-1"
    />
    <div
      className="listHolder col-10"
    >
      <h5
        className="listHeader"
      >
         Pending 
      </h5>
    </div>
    <div
      className="col-1"
    />
  </div>
  <div
    className="row"
  >
    <div
      className="col-1"
    />
    <div
      className="listHolder col-10"
      id="CompletedList"
    >
      <h5
        className="listHeader"
      >
         Completed 
      </h5>
    </div>
    <div
      className="col-1"
    />
  </div>
</div>
`;
